buildSrc/src/main/kotlin/VersionUtils.kt

```diff
@@ -1,29 +1,51 @@
 /**
- * Get the versionCode for a given semver
+ * Get the version code for a given semantic version.
+ * Does not validate the input and thus will throw an exception when parts are missing.
  *
- * Sample output:
- * 0.0.0 -> 0
- * 1.1.1 -> 10101
- * 0.7.0 -> 700
- * 99.99.99 -> 999999
+ * The pre-release part ("-rc.1", "-beta.1" etc.) defaults to 99
  *
- * @return the versionCode, or null if value is invalid.
+ * Sample output:
+ * MA.MI.PA-PR   -> MAMIPAPR
+ * 0.0.0         ->       99
+ * 1.1.1         ->  1010199
+ * 0.7.0         ->    70099
+ * 99.99.99      -> 99999999
+ * 2.0.0-rc.3    ->  2000003
+ * 2.0.0         ->  2000099
+ * 99.99.99-rc.1 -> 99999901
  */
 fun getVersionCode(versionName: String): Int? {
-    val parts = versionName
-        .substringBefore('-')
+    // Split to core and pre release parts with a default for pre release (null)
+    val (versionCore, versionPreRelease) = versionName
+        .split("-", limit = 2)
+        .let {
+            when (it.size) {
+                // No pre release part included
+                1 -> arrayOf(it[0], null)
+                // Pre release part included
+                else -> arrayOf(it[0], it[1])
+            }
+        }
+
+    // Parse core part
+    val (major, minor, patch) = versionCore!!
         .splitToSequence('.')
         .mapNotNull(String::toIntOrNull)
         .take(3)
         .toList()
 
-    // Not a valid semver
-    if (parts.size != 3) return null
+    // Parse pre release part (ignore type, only get the number)
+    val buildVersion = versionPreRelease
+        ?.split('.', limit = 2)
+        ?.getOrNull(1)
+        ?.let(String::toIntOrNull)
```

Comment:
This should have better performance and readability:
```suggestion
    val buildVersion = versionPreRelease?.run {
        substringAfter('.').let(String::toIntOrNull)
    }
```